sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/DownloadCBDE.targets

```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- Targets to download the CBDE binaries if required. 
+  
+       The files will be downloaded automatically if they are not found locally.
+       
+       To force the files to be downloaded even if they exist, set the build property "UpdateCBDE" to true.
+  
+  -->
+
+  <PropertyGroup>
+    <RelativeCBDEFolder>CBDE\</RelativeCBDEFolder>
+    <CBDEFolder>$(MSBuildProjectDirectory)\$(RelativeCBDEFolder)</CBDEFolder>
+    <SpecificCBDEVersion>1.0.0.13341</SpecificCBDEVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <CBDEUrls Include="http://repox.jfrog.io/repox/sonarsource-public-builds/org/sonarsource/cbde/$(SpecificCBDEVersion)/cbde-windows.$(SpecificCBDEVersion).zip" >
+      <SubFolder>windows\</SubFolder>
+      <MainExe>dotnet-symbolic-execution.exe</MainExe>
+      <ZipFileName>cbde-windows.zip</ZipFileName>
+    </CBDEUrls>
+    <!-- Only Windows is supported for now
+    <CBDEUrls Include="http://repox.jfrog.io/repox/sonarsource-public-builds/org/sonarsource/cbde/$(SpecificCBDEVersion)/cbde-linux.$(SpecificCBDEVersion).zip" >
+      <SubFolder>linux\</SubFolder>
+      <MainExe>dotnet-symbolic-execution</MainExe>
+      <ZipFileName>cbde-linux.zip</ZipFileName>
+    </CBDEUrls>
+    -->
+  </ItemGroup>
+
+
+  <!-- Specify them here in case they are already present. If they are not, they will be added by the task that fetches the files -->
+  <!-- This is a duplication of the "Content" items defined inside the target FetchCBDEBinaries, and we should try to remove it.
+       However, if we simply remove it, when running unit tests from the IDE while SonarAnalyzer.CSharp is already up-to-date
+       these files are missing. Duplication the item definition here avoid this situation
+       -->
+  <ItemGroup>
+    <Content Include="$(RelativeCBDEFolder)\windows\**\*.exe" >
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="$(RelativeCBDEFolder)\windows\**\*.txt" >
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+    <Content Include="$(RelativeCBDEFolder)\linux\**" Exclude="$(RelativeCBDEFolder)\linux\**\*.zip">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+
+  
+
+
+  <Target Name="FetchCBDEBinaries" BeforeTargets="BeforeBuild" Outputs="%(CBDEUrls.SubFolder)/%(CBDEUrls.MainExe)">
+    <PropertyGroup>
+      <PowerShellExe Condition="'$(PowerShellExe)'==''">%WINDIR%\System32\WindowsPowerShell\v1.0\powershell.exe</PowerShellExe>
+      <UnzipScript>$(MSBuildProjectDirectory)\..\..\..\Scripts\unzip.ps1</UnzipScript>
```

Comment:
there are built-in MSBuild tasks for 
- Download https://docs.microsoft.com/en-us/visualstudio/msbuild/downloadfile-task?view=vs-2019 
- Unzip https://docs.microsoft.com/en-us/visualstudio/msbuild/unzip-task?view=vs-2019

I think it's better to use that rather then maintaining a set of scripts

Reply:
See #2885